### PR TITLE
feature/CU-jv46zt_FRONT-END-Create-quiz-results-page

### PR DESCRIPTION
**Pull request by Thurstag**

### CODE REVIEW

**Comment by Thurstag** [comment]
Task linked: [CU-jv46zt [FRONT-END] Create quiz results page](https://app.clickup.com/t/jv46zt)

**Comment by Thurstag** [comment]
Task linked: [CU-jv46zt [FRONT-END] Create quiz results page](https://app.clickup.com/t/jv46zt)

**Comment by Thurstag** [comment]
image

**Comment by Fantorem** [comment]
Test la taille

File: lib/model/api/quiz.dart [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    return StringMultipleChoiceQuestionFail.fromJSON(json);
+  }
+}
+
+/// Question fail for a StringMultipleChoiceQuestion
+class StringMultipleChoiceQuestionFail extends AQuestionFail {
+  static const String USER_ANSWER_FIELD_NAME = 'userAnswer';
+  static const String SOLUTION_FIELD_NAME = 'solution';
+
+  final String userAnswer;
```
Comment (Thurstag) [inline lib/model/api/quiz.dart]: `final String? userAnswer;`

File: lib/model/api/quiz.dart [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  final String solution;
+
+  /// Constructor
+  StringMultipleChoiceQuestionFail(String userAnswer, String solution)
+      : userAnswer = userAnswer,
+        solution = solution;
+
+  /// Load a StringMultipleChoiceQuestionFail from JSON
+  factory StringMultipleChoiceQuestionFail.fromJSON(Map<String, dynamic> json) {
+    if (!json.containsKey(USER_ANSWER_FIELD_NAME) ||
```
Comment (Thurstag) [inline lib/model/api/quiz.dart:126]:
```
if (!json.containsKey(USER_ANSWER_FIELD_NAME) ||
        !json.containsKey(SOLUTION_FIELD_NAME) ||
        (json[USER_ANSWER_FIELD_NAME] != null &&
            !(json[USER_ANSWER_FIELD_NAME] is String)) ||
        !(json[SOLUTION_FIELD_NAME] is String)) {
      throw Exception('Malformed StringMultipleChoiceQuestionFail');
    }
```
\+ Enrichir les tests

File: lib/pages/solo/quiz_results.dart [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            ),
+          ),
+        ),
+        Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    generateTextCard(context, 'User', Colors.blue),
```
Comment (Thurstag) [inline lib/pages/solo/quiz_results.dart]: Ne pas afficher si pas de fails

**Review by Tomansion — APPROVED** [review]

**Review by YesLife — APPROVED** [review]

**Review by Fantorem — APPROVED** [review]

**Pull request merged**